### DISCUSSION TITLE
Query data in to CSV file

### DISCUSSION THREAD

**Discussion by vmotakat** [GRIDAPPSD/gridappsd-forum#116] [API] [answered]
how to get following query data of given test feeder in to CSV format ( I am able to see from blaze graph but I need same in csv file)

# into an OPTIONAL block.
#
# NOTE: Naming conventions: "inverter_" is associated with the
# "Wires::PowerElectronicsConnection" object, and "phase_" is
# associated with the "Wires::PowerElectronicsConnectionPhase"
# object. Documentation:
# https://gridappsd.readthedocs.io/en/latest/developer_resources/index.html#cim-documentation
PREFIX r: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX c: <http://iec.ch/TC57/CIM100#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
SELECT ?inverter_mrid ?inverter_name ?inverter_mode ?inverter_max_q ?inverter_min_q ?inverter_p ?inverter_q ?inverter_rated_s ?inverter_rated_u ?phase_mrid ?phase_name ?phase_p ?phase_q (group_concat(distinct ?phs;separator="\n") as ?phases) 
WHERE {
# Update for your feeder, or remove for all feeders.
VALUES ?feeder_mrid {"_AAE94E4A-2465-6F5E-37B1-3E72183A4E44"}
?s r:type c:PowerElectronicsConnection.
?s c:Equipment.EquipmentContainer ?fdr.
?fdr c:IdentifiedObject.mRID ?feeder_mrid.
?s c:IdentifiedObject.mRID ?inverter_mrid.
?s c:IdentifiedObject.name ?inverter_name.
?s c:PowerElectronicsConnection.p ?inverter_p.
?s c:PowerElectronicsConnection.q ?inverter_q.
?s c:PowerElectronicsConnection.ratedS ?inverter_rated_s.
?s c:PowerElectronicsConnection.ratedU ?inverter_rated_u.
OPTIONAL {
?s c:PowerElectronicsConnection.inverterMode ?inverter_mode.
?s c:PowerElectronicsConnection.maxQ ?inverter_max_q.
?s c:PowerElectronicsConnection.minQ ?inverter_min_q.
}
OPTIONAL {
?pecp c:PowerElectronicsConnectionPhase.PowerElectronicsConnection ?s.
?pecp c:IdentifiedObject.mRID ?phase_mrid.
?pecp c:IdentifiedObject.name ?phase_name.
?pecp c:PowerElectronicsConnectionPhase.p ?phase_p.
?pecp c:PowerElectronicsConnectionPhase.q ?phase_q.
?pecp c:PowerElectronicsConnectionPhase.phase
?phsraw bind(strafter(str(?phsraw),"SinglePhaseKind.") as ?phs)
}
}
GROUP BY ?inverter_mrid ?inverter_name ?inverter_rated_s ?inverter_rated_u ?inverter_p ?inverter_q ?phase_mrid ?phase_name ?phase_p ?phase_q ?inverter_mode ?inverter_max_q ?inverter_min_q
ORDER BY ?inverter_mrid

@https://github.com/AAndersn

**Comment by AAndersn** [comment]
There is a known issue #81 that PowerGrid Models API returns an empty dict object if CSV is specified as the result format

**Comment by AAndersn** [comment]
Screenshot of issue from #81 with a custom SPARQL string:

JSON output
![image](https://user-images.githubusercontent.com/19935503/163627982-3779837c-8653-4030-85ca-19e330a4f9ed.png)
CSV output
![image](https://user-images.githubusercontent.com/19935503/163627841-0cff5d95-de74-46a6-96e7-96e1793208de.png)

**Comment by poorva1209** [comment]
@vmotakat You can request the data in JSON format and use python to convert that to CSV. There are many JSON to CSV convert examples you can find online using python3. This way you are not blocked by issue #81

**Comment by vmotakat** [comment]
Thank you !!

**Comment by poorva1209** [comment, marked as answer]
@vmotakat There is also an 'export' button on blazegraph web interface. After executing the query you can directly make a csv from there. I think that would be fastest and easiest.

**Comment by vmotakat** [comment]
can you tell me where I can find export option in blazegraph web interface?

**Comment by poorva1209** [comment]
Execute the query and then scroll down to the bottom of the QUERY web page. It is on the right-hand side at the bottom. 
![image](https://user-images.githubusercontent.com/6538501/163630994-9a1c192d-0e7d-46e4-91ff-ce064c4f5364.png)

**Comment by vmotakat** [comment]
ok thank you. I got CSV file.
